### PR TITLE
coll/basic: Fix clang warnings

### PR DESCRIPTION
**Pull request by jjhursey**
* Possible memory leak of tmp variable
 * `err` variable not checked or used

### CODE REVIEW

File: ompi/mca/coll/basic/coll_basic_reduce_scatter_block.c [hunk 1]
```diff
@@ -105,8 +106,11 @@ mca_coll_basic_reduce_scatter_block_inter(const void *sbuf, void *rbuf, int rcou
         span = opal_datatype_span(&dtype->super, totalcounts, &gap);
 
         tmpbuf = (char *) malloc(span);
+        if (NULL == tmpbuf) {
+            return OMPI_ERR_OUT_OF_RESOURCE;
+        }
         tmpbuf2 = (char *) malloc(span);
-        if (NULL == tmpbuf || NULL == tmpbuf2) {
+        if (NULL == tmpbuf2) {
             return OMPI_ERR_OUT_OF_RESOURCE;
```
Comment (bosilca) [inline ompi/mca/coll/basic/coll_basic_reduce_scatter_block.c:114]: We are leaking tmp here. I would rewrite the code to handle both buffers into a single allocation.
```c
tmpbuf = (char*)malloc(2*span);
if(NULL == tmpbuf) {
    return OMPI_ERR_OUT_OF_RESOURCE;
}
tmpbuf2 = tmpbuf + span;
```
Maybe with a realignment of tmp2 to 8 bytes.

Comment (jjhursey) [reply ompi/mca/coll/basic/coll_basic_reduce_scatter_block.c:114]: Doh. Can't believe I forgot the free as that was the whole point. I changed it to a single allocation.

**Comment by jjhursey** [comment]
@bosilca Can you re-review?

**Comment by jjhursey** [comment]
@bosilca ping

**Review by bosilca — APPROVED** [review]

**Pull request merged**